engines/dfc_provider/app/controllers/dfc_provider/application_controller.rb

```diff
@@ -8,6 +8,7 @@ class ApplicationController < ActionController::Base
     protect_from_forgery with: :null_session
 
     rescue_from ActiveRecord::RecordNotFound, with: :not_found
+    rescue_from CanCan::AccessDenied, with: :unauthorized
```

Comment:
Do none of the existing dfc controllers check authorisation already? Maybe they should?!

I'm guessing that most/all of the api read and write actions already filter records according to the user's enterprises, but is it worth covering the potential holes... or am I missing something?

Reply:
We haven't used CanCan for these checks until now. As you guessed, we used the users's scopes to access the right data.

I personally dislike CanCan as it adds complexity and I always find it difficult to see which rules apply to the current user. Looking for records in the right scopes makes any further authorisation unnecessary and it seems more efficient to me.

That said, sometimes the authorisation step is a bit more complicated and when you need to perform the same in several places then it's good to have a single source of truth. CanCan does have its place there.

Spree also used it to check every action and you had to allow everything with abilities before a user could access it. That's good security practice but it made the ability model unnecessarily big, I think.

Is this worth a chat in our next meeting?